app/src/main/java/com/ayush/weatherapp/main/MainActivity.java

```diff
@@ -1,15 +1,80 @@
 package com.ayush.weatherapp.main;
 
 import android.os.Bundle;
-import android.support.v7.app.AppCompatActivity;
+import android.support.design.widget.NavigationView;
+import android.support.v4.view.GravityCompat;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBar;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.Toast;
+import butterknife.BindView;
+import com.ayush.weatherapp.BaseActivity;
 import com.ayush.weatherapp.R;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends BaseActivity {
+
+  @BindView(R.id.layout_drawer) DrawerLayout drawerLayout;
+  @BindView(R.id.nav_view) NavigationView navigationView;
+  @BindView(R.id.toolbar) Toolbar toolbar;
+
+  @Override protected int getContextView() {
+    return R.layout.navigation_drawer;
+  }
 
   @Override protected void onCreate(Bundle savedInstanceState) {
     setTheme(R.style.AppTheme);
 
     super.onCreate(savedInstanceState);
-    setContentView(R.layout.main);
+
+    setActionBar();
+
+    setNavigationView();
+  }
+
+  @Override public boolean onCreateOptionsMenu(Menu menu) {
+    MenuInflater inflater = getMenuInflater();
+    inflater.inflate(R.menu.menu_main, menu);
+    return true;
+  }
+
+  @Override public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case android.R.id.home:
+        drawerLayout.openDrawer(GravityCompat.START);
+        return true;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  private void setActionBar() {
+    setSupportActionBar(toolbar);
+
+    ActionBar actionbar = getSupportActionBar();
+
+    if (actionbar != null) {
```

Comment:
the null check is not required for action bars it's either there not not there

Reply:
:ok: